newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackInformationPage.js

```diff
@@ -255,46 +313,147 @@ const PrivateAssetPackInformationPage = ({
     ]
   );
 
-  const getBuyButton = i18n => {
+  const onPurchaseWithCredits = React.useCallback(
+    async (i18n: I18nType) => {
+      if (!privateAssetPackListingData || !assetPack) return;
+
+      if (!profile || !limits) {
+        // User not logged in, suggest to log in.
+        onOpenLoginDialog();
+        return;
+      }
+
+      if (isAlreadyReceived) {
+        onAssetPackOpen(privateAssetPackListingData);
+        return;
+      }
+
+      sendAssetPackBuyClicked({
```

Comment:
Should we specify the currency? (Money vs Credits)

Reply:
you're right